ContainerViewController/Source/ContainerViewController.swift

```diff
@@ -0,0 +1,189 @@
+        super.viewDidAppear(animated)
+        isOnScreen = true
+    }
+
+    public override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        isOnScreen = false
+    }
+
+    /// Performs transition to the new view controller
+    ///
+    /// - Parameters:
+    ///   - controller: The view controller that will be presented
+    ///   - style: The type of transition animation
+    ///   - animated: Perform or do not perform animation during transition
+    public final func present(_ controller: UIViewController, style: TransitionStyle, animated: Bool) {
+        guard controller !== currentViewController else {
+            return
+        }
+        willPresentViewController(controller, animated: animated)
+        let actuallyOnScreen: Bool = isOnScreen
+        let oldController: UIViewController? = currentViewController
+        currentViewController = controller
+
+        oldController?.willMove(toParent: nil)
+        if actuallyOnScreen {
+            oldController?.beginAppearanceTransition(false, animated: animated)
+        }
+        addChild(controller)
+        if actuallyOnScreen {
+            controller.beginAppearanceTransition(true, animated: animated)
+        }
+
+        let fromView: UIView? = oldController?.view
+
+        let toView: UIView = controller.view
+        toView.translatesAutoresizingMaskIntoConstraints = true
+        toView.frame = containerView.bounds
+        toView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        containerView.addSubview(toView)
+
+        view.isUserInteractionEnabled = false
+        let finishTransition: () -> Void = {
+            if let actualOld: UIViewController = oldController {
+                actualOld.view.removeFromSuperview()
+                if actuallyOnScreen {
+                    actualOld.endAppearanceTransition()
+                }
+                actualOld.removeFromParent()
+            }
+            if actuallyOnScreen {
+                controller.endAppearanceTransition()
+            }
+            controller.didMove(toParent: self)
+            self.view.isUserInteractionEnabled = true
+            self.didPresentViewController(controller, animated: animated)
+        }
+
+        if animated {
+            switch (style, fromView) {
+            case (.push, .some(let actualFromView)):
+                animatePushPop(fromView: actualFromView, toView: toView, push: true, didFinish: finishTransition)
+            case (.pop, .some(let actualFromView)):
+                animatePushPop(fromView: actualFromView, toView: toView, push: false, didFinish: finishTransition)
+            case (.fade, _):
+                animateFade(fromView: nil, toView: toView, didFinish: finishTransition)
+            case (.custom(let animator), _):
+                animator.animate(fromView: fromView, toView: toView, containerView: containerView, didFinish: finishTransition)
+            default:
```

Comment:
Shouldn't here be an assert about "view is nil"?

Reply:
No. fromView can be nil, for example for the first time.